test/framework/clean_up.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
+// CleanUpInput are all the dependencies needed to clean up a Cluster API cluster.
+type CleanUpInput struct {
+	Management    ManagementCluster
+	Cluster       *corev1.Cluster
+	DeleteTimeout time.Duration
+}
+
+func (c *CleanUpInput) SetDefaults() {
+	if c.DeleteTimeout == 0*time.Second {
+		c.DeleteTimeout = 2 * time.Minute
+	}
+}
+
+// CleanUp deletes the cluster and waits for everything to be gone.
+// Generally this test can be reused for many tests since the implementation is so simple.
+func CleanUp(input *CleanUpInput) {
+	// TODO: check that all the things we expect have the cluster label or else they can't get deleted
+	input.SetDefaults()
+	ctx := context.Background()
+	mgmtClient, err := input.Management.GetManagementClient()
+	Expect(err).NotTo(HaveOccurred(), "stack: %+v", err)
+
+	Expect(mgmtClient.Delete(ctx, input.Cluster)).NotTo(HaveOccurred())
+
+	Eventually(func() []corev1.Cluster {
+		clusters := corev1.ClusterList{}
+		c, err := input.Management.GetManagementClient()
```

Comment:
Maybe I'm missing something, don't see `GetManagementClient()` defined in the `ManagementCluster` interface

Reply:
I made a mistake in the code you reviewed, it's updated to GetClient properly now.